### PR TITLE
Added Google authentication using Firebase

### PR DESCRIPTION
**Pull request by SachinRammoorthy**
Google authentication using Firebase. Also updated documentation.
If the user has already signed in once, he/she is automatically signed in when they visit the page again. 

Screenshots:
screen shot 2017-12-21 at 9 04 15 pm
screen shot 2017-12-25 at 10 45 31 pm
screen shot 2017-12-21 at 9 04 02 pm

### CODE REVIEW

File: public/js/script.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          document.getElementById('prof_pic').style.visibility = "hidden";
+
+        }
+        document.getElementById('quickstart-sign-in').disabled = false;
+      });
+      document.getElementById('quickstart-sign-in').addEventListener('click', toggleSignIn, false);
+    }
+    window.onload = function() {
+      initApp();
+    };
```
Comment (neelanshsahai) [inline public/js/script.js]: please add a blank line here

File: views/index.jade [hunk 2]
```diff
@@ -2,4 +2,4 @@ extends layout
 
 block content
   h1= title
-  p Welcome to #{title}
+  p Welcome to #{title}
```
Comment (neelanshsahai) [inline views/index.jade]: Another blank line here as well.

**Comment by SachinRammoorthy** [comment]
@neelanshsahai I have updated the PR. Please have a look.

**Comment by garg-anuj** [comment]
Awesome work

**Pull request closed without merging**